src/Generic/Data/Internal/Generically.hs

```diff
@@ -33,13 +38,27 @@ instance (Generic a, Ord (Rep a ())) => Ord (Generically a) where
 instance (Generic a, GShow0 (Rep a)) => Show (Generically a) where
   showsPrec = gshowsPrec
 
-instance (Generic a, Semigroup (Rep a ())) => Semigroup (Generically a) where
+type SemigroupAssertNoSum a =
+    Assert (Not (HasSum (Rep a)))
+    ('Text "No generic instances of " ':<>: 'ShowType Semigroup ':<>: 'Text " for sum types")
```

Comment:
Moving this definition into the `Error` module would allow keeping fewer imports and extensions here.